CriteoPublisherSdk/Sources/Standalone/CRBannerView.m

```diff
@@ -107,11 +129,15 @@ - (void)loadAd {
   CR_CacheAdUnit *cacheAdUnit = [[CR_CacheAdUnit alloc] initWithAdUnitId:_adUnit.adUnitId
                                                                     size:self.frame.size
                                                               adUnitType:CRAdUnitTypeBanner];
-  CR_CdbBid *bid = [self.criteo getBid:cacheAdUnit];
-
-  if ([bid isEmpty]) return [self safelyNotifyAdLoadFail:CRErrorCodeNoFill];
 
-  [self loadAdWithDisplayData:bid.displayUrl];
+  [self.criteo getBid:cacheAdUnit
+      responseHandler:^(CR_CdbBid *bid) {
+        if (!bid || bid.isEmpty) {
+          [self safelyNotifyAdLoadFail:CRErrorCodeNoFill];
```

Comment:
nit: `notifyAdLoadFailSafely` (to conserve the convention of starting methods/functions with verbs)

Reply:
Same comment as previous